File: offer57-II-he-wei-sde-lian-xu-zheng-shu-xu-lie-lcof/javascript/Solution.js
/**
 * @param {number} target
 * @return {number[][]}
 */
var findContinuousSequence = function(target) {
    let result = []
    let p = 1
    let q = 2
    let sum = 3
    while (p < q) {
        if (sum === target) {
            let arr = new Array(q-p+1)
            for (let i = p; i <= q; ++i) {
                arr[i-p] = i
            }
            result.push(arr)
            sum -= p
            p++
            q++
            sum += q
        } else if (sum > target) {
            sum -= p
            p++
        } else {
            q++
            sum += q
        }
    }
    return result
};

File: offer57-II-he-wei-sde-lian-xu-zheng-shu-xu-lie-lcof/python/Solution.py
class Solution:
    def findContinuousSequence(self, target: int) -> List[List[int]]:
        result = []
        p = 1
        q = 2
        total = 3
        while p < q:
            arr = [0] * (q - p + 1)
            if total == target:
                for i in range(p, q+1):
                    arr[i-p] = i
                result.append(arr)
                total -= p
                p += 1
                q += 1
                total += q
            elif total > target:
                total -= p
                p += 1
            else:
                q += 1
                total += q
        return result
